lib/teleterm/clusters/cluster_file_transfer.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusters
+
+import (
+	"io"
+	"os"
+	"time"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/sshutils/sftp"
+	api "github.com/gravitational/teleport/lib/teleterm/api/protogen/golang/v1"
+
+	"github.com/gravitational/trace"
+)
+
+func (c *Cluster) TransferFile(request *api.FileTransferRequest, server api.TerminalService_TransferFileServer) error {
```

Comment:
I noticed that stream errors are returned as UNKNOWN rather than a specific gRPC status code.

![unknown error](https://user-images.githubusercontent.com/27113/194021791-8749930c-0096-41da-8757-d25d3f94593e.png)

That's because `apiserver.withErrorHandling` adds a `grpc.UnaryServerInterceptor` which converts Go errors to gRPC status codes. However, we'd need to add a specific interceptor for streams so that the errors are converted there too.

I noticed that first after Lisa added the stream for passwordless login.

As long as the frontend doesn't depend on the status code of the error, the issue is manageable. I'll try to add an issue for tracking it this week.

Reply:
Good catch, I didn't realize that `UNKNOWN` shows up because of this.